src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala

```diff
@@ -1,61 +1,66 @@
 package org.clulab.habitus.utils
 
-import org.clulab.habitus.variables.EntityDistFreq
 import org.clulab.odin.{Attachment, Mention}
 import org.clulab.processors.{Document, Sentence}
 import org.clulab.struct.Interval
-import scala.util.control.Breaks._
 
-import scala.collection.{breakOut, mutable}
-import scala.collection.mutable.ArrayBuffer
+import scala.collection.immutable.ListMap
+
 
 trait Context extends Attachment with Pairable
 
-case class DefaultContext(location: String, date: String, process: String, crop: String, fertilizer: String, comparative: Int) extends Context
+case class DefaultContext(location: String, date: String, process: String, crop: String, fertilizer: String, season:String, comparative: Int) extends Context
 
 case class BeliefContext(context: String) extends Context
 
 trait ContextExtractor {
 
   val NA = "N/A"
   val maxContextWindow = 2
+  val processToLemmas = ListMap(
+    "planting"         -> Set("plant", "sow", "cultivate", "cultivation", "grow", "seed", "seeding", "seedling", "transplant", "cropping", "variety", "use", "growth"),
+    "harvesting"       -> Set("harvest", "yield"),
+    "credit"           -> Set("credit", "finance", "value"),
+    "irrigation"       -> Set("irrigation", "irrigate"),
+    "weeds"            -> Set("weed"),
+    "natural_disaster" -> Set("flood", "bird", "attack", "floodwater"),
+    "fertilizerApplication" -> Set("fertilizer", "application", "apply", "compost", "rate", "concentration"),
+    "climate" -> Set("climate"),
+    "agriculture" -> Set("agriculture"),
+    NA -> Set("N/A")
+  )
+
+  def reverseProcessToLemma(processToLemma: ListMap[String, Set[String]]): Map[String, String] = {
+    val lemmaToProcess = collection.mutable.Map[String, String]()
+    for ( (process, lemmaValues) <- processToLemma){
+      for (lemmaValue <- lemmaValues){
+        lemmaToProcess += (lemmaValue -> process)
+      }
+    }
+    lemmaToProcess.toMap
+  }
 
-  val plantingLemmas: Seq[String] = Seq("plant", "sow", "cultivate", "cultivation", "grow")
-  val creditLemmas: Seq[String] = Seq("credit", "finance", "value")
-  val harvestLemmas: Seq[String] = Seq("harvest", "yield")
-  val irrigationLemmas: Seq[String] = Seq("irrigation", "irrigate")
-  val weedsLemmas: Seq[String] = Seq("weed")
-  val disasterLemmas: Seq[String] = Seq("flood", "bird", "attack")
+  val lemmaToProcess: Map[String, String] = reverseProcessToLemma(processToLemmas)
 
   def getContextPerMention(mentions: Seq[Mention], doc: Document): Seq[Mention]
 
   def getProcess(mention: Mention): String = {
+    val argLabels = mention.arguments.flatMap(_._2).map(_.label).toSeq
+    if (argLabels.contains("Fertilizer")) {
+      "fertilizerApplication"
+    } else {
+      //    Getting process lemma from the mention
+      val mentionLemmas = mention.lemmas
+      val mensLemmaOverLap = processToLemmas.map(p => (p._1, p._2.intersect(mentionLemmas.get.toSet).toList.length))
+      if (mensLemmaOverLap.values.max != 0) {
```

Comment:
It looks like `mensLemmaOverLap.values.max` can be taken out of the filter loop.